src/process/receipts.rs

```diff
@@ -0,0 +1,297 @@
+                    .eq(schema::receipts::dsl::receipt_id),
+            ),
+        )
+        .filter(
+            schema::execution_outcome_receipts::dsl::receipt_id.eq_any(
+                receipts
+                    .iter()
+                    .map(|r| r.receipt_id.to_string())
+                    .collect::<Vec<_>>(),
+            ),
+        )
+        .select((
+            schema::execution_outcome_receipts::dsl::receipt_id,
+            schema::receipts::dsl::transaction_hash,
+        ))
+        .load_async(&pool)
+        .await;
+
+    let tx_hashes_for_receipt_via_transactions: Result<
+        Vec<(String, String)>,
+        tokio_diesel::AsyncError,
+    > = schema::transactions::table
+        .filter(
+            schema::transactions::dsl::receipt_id.eq_any(
+                receipts
+                    .iter()
+                    .map(|r| r.receipt_id.to_string())
+                    .collect::<Vec<_>>(),
+            ),
+        )
+        .select((
+            schema::transactions::dsl::receipt_id,
+            schema::transactions::dsl::transaction_hash,
+        ))
+        .load_async(&pool)
+        .await;
+
+    let mut tx_hashes_for_receipts: HashMap<String, String> = HashMap::new();
+
+    if let Ok(result) = tx_hashes_for_receipts_via_outcomes {
+        tx_hashes_for_receipts.extend(result);
+    }
+
+    if let Ok(result) = tx_hashes_for_receipt_via_transactions {
+        tx_hashes_for_receipts.extend(result);
+    }
+
+    let receipt_models: Vec<models::receipts::Receipt> = receipts
+        .iter()
+        .filter_map(|r| {
+            if tx_hashes_for_receipts.contains_key(r.receipt_id.to_string().as_str()) {
+                Some(models::Receipt::from_receipt_view(
+                    r,
+                    block_height,
+                    tx_hashes_for_receipts
+                        .get(r.receipt_id.to_string().as_str())
+                        .expect("transaction hash expected as we have check for it earlier")
+                        .clone(),
+                ))
+            } else {
+                warn!(
+                    target: crate::INDEXER_FOR_EXPLORER,
+                    "Skipping Receipt {} as we can't find parent Transaction for it.",
+                    r.receipt_id.to_string()
+                );
+                skipping_receipt_ids.push(r.receipt_id);
+                None
+            }
+        })
+        .collect();
+
+    let save_receipts_future = save_receipts(&pool, receipt_models);
+
+    let receipts_to_process: Vec<&near_indexer::near_primitives::views::ReceiptView> = receipts
+        .into_iter()
+        .filter(|r| !skipping_receipt_ids.contains(&r.receipt_id))
+        .collect();
+
+    let process_receipt_actions_future = process_receipt_actions(&pool, &receipts_to_process);
+
+    let process_receipt_data_future = process_receipt_data(&pool, &receipts_to_process);
```

Comment:
Consider using [`Iterator::partition()`](https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.partition) to have two separate vectors with (1) Action Receipts and (2) Data Receipts only.